### PR TITLE
fix metric

### PR DESCRIPTION
**Pull request by JianGuanTHU**
**Description:**

**Reference Issues:** #126

### CODE REVIEW

File: cotk/metric/metric.py [hunk 1]
```diff
@@ -63,9 +63,7 @@ def forward(self, data):
 			data (dict): A dict at least contains the following keys.
 			data[reference_allvocabs_key] (list of list of list): Reference sentences.
 				Does not contain start token (eg: ``<go>``) and end token (eg: ``<eos>``).
-				Outermost list: batch_size
-				Innermost list: number of words, allow different sizes
-				Second innermost list: number of sentences, allow different sizes
+				Size: `[batch_size, sentence_num, max_sentence_length]`
```
Comment (hzhwcmhf) [inline cotk/metric/metric.py]: It's important to claim that the 2nd and 3rd dimensions can be different sizes. It's different from a matrix with fixed length

Comment (hzhwcmhf) [reply cotk/metric/metric.py]: Maybe you can add a example for that. Find tensorflow or pytorch for reference

Comment (JianGuanTHU) [reply cotk/metric/metric.py]: got it

File: cotk/metric/metric.py [hunk 2]
```diff
@@ -63,9 +63,11 @@ def forward(self, data):
 			data (dict): A dict at least contains the following keys.
 			data[reference_allvocabs_key] (list of list of list): Reference sentences.
 				Does not contain start token (eg: ``<go>``) and end token (eg: ``<eos>``).
-				Outermost list: batch_size
-				Innermost list: number of words, allow different sizes
-				Second innermost list: number of sentences, allow different sizes
+				First dimension: batch_size
+				Second dimmension: number of sentences, allow different sizes for different batches
+				Third dimmension: number of words, allow different sizes for different sentences
```
Comment (hzhwcmhf) [inline cotk/metric/metric.py]: Maybe you can use the expression like [batch_size, ~sent_num, ~word_num] and say '~' means different size

File: cotk/metric/metric.py [hunk 3]
```diff
@@ -508,24 +529,25 @@ def close(self):
 					bleu_irl.append(self.run_f((ref[:i]+ref[i+1:], ref[i])))
 			return {"self-bleu" : 1.0 * sum(bleu_irl) / len(bleu_irl)}
 		except ZeroDivisionError as _:
```
Comment (hzhwcmhf) [inline cotk/metric/metric.py]: delete the exception
